Providers/Profile.swift

```diff
@@ -445,6 +445,13 @@ public class BrowserProfile: Profile {
                 if oldValue == syncLock {
                     return
                 }
+
+                if syncLock == 0 {
```

Comment:
Musing: could we not just attach this as a listener to `ProfileDidFinishSyncingNotification`, so we decouple this?